deltacat/compute/compactor_v2/deletes/delete_strategy_equality_delete.py

```diff
@@ -0,0 +1,163 @@
+            Apply delete operations on the given table using all provided delete file envelopes.
+    """
+
+    _name = "EqualityDeleteStrategy"
+
+    @property
+    def name(self) -> str:
+        """
+        The name of the delete strategy.
+        """
+        return self._name
+
+    def _drop_rows(
+        self,
+        table: pa.Table,
+        delete_table: pa.Table,
+        delete_column_names: List[str],
+        equality_predicate_operation: Optional[Callable] = pa.compute.and_,
+    ) -> Tuple[pa.Table, int]:
+        """
+        Drop rows from the given table based on the provided delete table and column names.
+
+        Args:
+            table (pa.Table): The input table to drop rows from.
+            delete_table (pa.Table): The table containing the values to match for deletion.
+            delete_column_names (List[str]): A list of column names to check for equality with the delete table.
+            equality_predicate_operation (Optional[Callable], optional): The operation to combine equality predicates for multiple columns.
+                Defaults to pa.compute.and_.
+
+        Returns:
+            Tuple[pa.Table, int]: A tuple containing the updated table after dropping rows,
+                and the number of rows dropped.
+        """
+        if len(delete_column_names) < 1:
+            return table, 0
+        # all 1s -> all True so wont discard any from the curr_boolean_mask
+        prev_boolean_mask = pa.array(np.ones(len(table), dtype=bool))
+        # all 0s -> all False so if mask is never modified all rows will be kept
+        curr_boolean_mask = pa.array(np.zeros(len(table), dtype=bool))
+        for delete_column_name in delete_column_names:
+            if delete_column_name not in table.column_names:
+                logger.warning(
+                    f"Column name {delete_column_name} not in table column names. Skipping dropping rows for this column."
+                )
+                continue
+            curr_boolean_mask = pc.is_in(
+                table[delete_column_name],
+                value_set=delete_table[delete_column_name],
+            )
+            curr_boolean_mask = equality_predicate_operation(
+                prev_boolean_mask, curr_boolean_mask
+            )
+            prev_boolean_mask = curr_boolean_mask
+        number_of_rows_before_dropping = len(table)
+        logger.debug(
+            f"Number of table rows before dropping: {number_of_rows_before_dropping}."
+        )
+        table = table.filter(pc.invert(curr_boolean_mask))
+        number_of_rows_after_dropping = len(table)
+        logger.debug(
+            f"Number of table rows after dropping: {number_of_rows_after_dropping}."
+        )
+        dropped_rows = number_of_rows_before_dropping - number_of_rows_after_dropping
+        return table, dropped_rows
+
+    def apply_deletes(
+        self,
+        table: Optional[pa.Table],
+        delete_file_envelope: DeleteFileEnvelope,
```

Comment:
Minor: If this method takes in a generic table to delete and columns to target, then I feel `apply_many_deletes` could easily re-use this  after concatenating all the tables.